process/transaction/shardProcess.go

```diff
@@ -789,6 +783,26 @@ func (txProc *txProcessor) processInnerTx(
 
 	txFee := txProc.computeInnerTxFee(innerTx)
 
+	innerTxHash := originalTxHash
+
+	var err error
+	if txProc.enableEpochsHandler.IsFlagEnabled(common.HashForInnerTransactionFlag) {
```

Comment:
Is this method called only for v3?

otherwise this will change also the behavior for SCRs for v1 and v2 relay txs, those need to be retested